.github/workflows/docs.yml

```
@@ -0,0 +1,92 @@
+name: Build and publish documentation
+on:
+  push:
+    # This prevents tag pushes from triggering this workflow
+    branches: ['**']
+  pull_request:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      publish-docs-website:
+        description: "Publish docs website to GitHub Pages?"
+        default: "false"
+jobs:
+  # =====================================================================================================================================================================
+  # Build documentation
+  # =====================================================================================================================================================================
+  build-documentation:
+    name: Build documentation
+    runs-on: ubuntu-latest
+    steps:
+      # ----------------------------------------------------------------------- Checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      # ----------------------------------------------------------------------- Set up tools
+      - name: Restore tool cache
+        id: tool-cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            doxygen-1.14.0.linux.bin.tar.gz
+            ubuntu_24.04_graphviz-13.0.0-cmake.deb
+          key: doxygen-1.14.0_graphviz-13.0.0
+
+      # We don't use apt to acquire Doxygen because the verison provided by Ubuntu is ancient
+      # For Graphviz the verison is old but still good enough, but it's actually significantly faster to just download it directly

```

```suggestion
      # For Graphviz the version is good enough, but still significantly faster to just download it directly
```

Simplified wording for future reference